experimental/devices/epd/epd.go

```diff
@@ -0,0 +1,463 @@
+		0x35, 0x51, 0x51, 0x19, 0x01, 0x00,
+	},
+	PartialUpdate: LUT{
+		0x10, 0x18, 0x18, 0x08, 0x18, 0x18, 0x08, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x13, 0x14, 0x44, 0x12,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	},
+}
+
+// DisplayConfig defines the options for the ePaper Device.
+type DisplayConfig struct {
+	W             int
+	H             int
+	FullUpdate    LUT
+	PartialUpdate LUT
+}
+
+// NewSPI returns a Dev object that communicates over SPI to a E-Paper display controller.
+func NewSPI(p spi.Port, dc, cs, rst gpio.PinOut, busy gpio.PinIO, config *DisplayConfig) (*Dev, error) {
+	if dc == gpio.INVALID {
+		return nil, errors.New("epd: use nil for dc to use 3-wire mode, do not use gpio.INVALID")
+	}
+
+	if err := dc.Out(gpio.Low); err != nil {
+		return nil, err
+	}
+
+	c, err := p.Connect(2*physic.MegaHertz, spi.Mode0, 8)
+	if err != nil {
+		return nil, err
+	}
+
+	d := &Dev{
+		c:      c,
+		dc:     dc,
+		cs:     cs,
+		rst:    rst,
+		busy:   busy,
+		lt:     Full,
+		config: config,
+		rect:   image.Rect(0, 0, config.W, config.H),
+	}
+
+	d.reset()
+
+	if err := d.sendCommand([]byte{byte(DriverOutputControl)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte((config.H - 1) & 0xFF)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(((config.H - 1) >> 8) & 0xFF)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(0x00)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendCommand([]byte{byte(BoosterSoftStartControl)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(0xD7)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(0xD6)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(0x9D)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendCommand([]byte{byte(WriteVcomRegister)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(0xA8)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendCommand([]byte{byte(SetDummyLinePeriod)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(0x1A)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendCommand([]byte{byte(SetGateTime)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(0x08)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendCommand([]byte{byte(DataEntryModeSetting)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.sendData([]byte{byte(0x03)}); err != nil {
+		return nil, err
+	}
+
+	if err := d.setLut(Full); err != nil {
+		return nil, err
+	}
+
+	return d, nil
+}
+
+// NewSPIHat returns a Dev object that communicates over SPI
+// and have the default config for the e-paper hat for raspberry pi
+func NewSPIHat(p spi.Port, config *DisplayConfig) (*Dev, error) {
+	dc := gpioreg.ByName(DataCommandPin)
+	cs := gpioreg.ByName(ChipSelectPin)
+	rst := gpioreg.ByName(ResetPin)
```

Comment:
You don't need to use the GPIO registry here, you can use use rpi.P1_11 for this pin, and the corresponding for others: https://periph.io/x/periph/host/rpi#pkg-variables

Thanks for separating the hat support from the device support.